packages/app-bridge/src/AppBridgePlatformApp.ts

```diff
@@ -15,27 +15,144 @@ import type {
     EventUnsubscribeFunction,
     StateAsEventName,
     StateReturn,
+    SubscribeMap,
 } from './AppBridge';
-import type { ApiMethodRegistry } from './registries/api/ApiMethodRegistry';
+import { Topic } from './types';
+import { ErrorMessageBus, IMessageBus, MessageBus } from './utilities/MessageBus';
+import { generateRandomString, notify, subscribe } from './utilities';
+import { getQueryParameters } from './utilities/queryParams';
+import type { ApiMethodRegistry } from './registries';
+import { openConnection } from './registries';
+import { InitializationError } from './errors';
 
 export type PlatformAppApiMethod = ApiMethodNameValidator<Pick<ApiMethodRegistry, 'getCurrentUser'>>;
 
-export type PlatformAppCommand = CommandNameValidator<Record<never, never>>;
+export type PlatformAppCommandRegistry = CommandNameValidator<{
+    openConnection: void;
+}>;
+
+export type PlatformAppCommand = CommandNameValidator<Pick<PlatformAppCommandRegistry, 'openConnection'>>;
 
 export type PlatformAppState = {
     settings: Record<string, unknown>;
 };
 
-export type PlatformAppContext = {
+type InitializeEvent = {
+    port: MessagePort;
+};
+
+type AppBaseProps = {
+    token: string;
     marketplaceServiceAppId: string;
+    connected: boolean;
+};
+
+export type PlatformAppContext = AppBaseProps & {
+    type: 'ASSET_ACTION';
+    assetId: string;
+    parentId: string;
+    directory: string;
```

Comment:
What is `directory`?

Reply:
Its the current directory where the Asset_action Platform App is triggered from.

Reply:
It will grow out of hand with the more `type` you add, be careful

Reply:
Absolutly agree!